libs/sysplugins/smarty_internal_compile_private_special_variable.php

```diff
@@ -94,9 +94,9 @@ public function compile($args, Smarty_Internal_TemplateCompilerBase $compiler, $
                         break;
                     }
                     if (strpos($_index[ 1 ], '$') === false && strpos($_index[ 1 ], '\'') === false) {
-                        return "@constant('{$_index[1]}')";
+                        return "defined('{$_index[1]}') ? constant('{$_index[1]}') : null";
```

Comment:
`return "defined('{$_index[1]}') ? constant('{$_index[1]}') : null";`
change to
`return "(defined('{$_index[1]}') ? constant('{$_index[1]}') : null)";`
??

Reply:
Useless braces are useless.

Reply:
I missed the fact that this is a code generation, in which case the braces may be necessary, given the fact the resulting expression may be used in ambiguous context..